### PR TITLE
Add `svelte/no-reactive-literals` rule

### PR DESCRIPTION
**Pull request by tivac**
Porting over another of the simpler rules from https://github.com/tivac/eslint-plugin-svelte/ like we discussed in https://github.com/ota-meshi/eslint-plugin-svelte/issues/198

### CODE REVIEW

File: src/rules/no-reactive-literals.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      // $: foo = 1;
+      [`${labeledStatementBase}[right.type="Literal"]`]: warn,
+
+      // $: foo = [];
+      [`${labeledStatementBase}[right.type="ArrayExpression"][right.elements.length=0]`]:
+        warn,
+
+      // $: foo = {};
+      [`${labeledStatementBase}[right.type="ObjectExpression"][right.properties.length=0]`]:
+        warn,
```
Comment (ota-meshi) [inline src/rules/no-reactive-literals.ts:66]: I think you can make it simpler by using `:matches()`.

```suggestion
      [`SvelteReactiveStatement > ExpressionStatement > AssignmentExpression:matches(${[
        // $: foo = "foo";
        // $: foo = 1;
        `[right.type="Literal"]`,
        // $: foo = [];
        `[right.type="ArrayExpression"][right.elements.length=0]`,
        // $: foo = {};
        `[right.type="ObjectExpression"][right.properties.length=0]`,
      ].join(",")})`]: warn,
```

https://github.com/estools/esquery#:~:text=matches%2Dany%3A-,%3Amatches,-(%5Battr%5D%20%3E%20%3Afirst%2Dchild

File: src/rules/no-reactive-literals.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+      return context.report({
+        node: parent,
+        loc: parent.loc,
+        messageId: "noReactiveLiterals",
+
+        fix(fixer) {
+          return [
+            // Insert "let" + whatever was in there
+            fixer.insertTextBefore(parent, `let ${source.getText(node)}`),
```
Comment (ota-meshi) [inline src/rules/no-reactive-literals.ts:46]: I think it's a difficult to make auto-fixes. If the variable has already been declared, you will get a syntax error.

```svelte
<script>
let foo;
$: foo = 42;
</script>
```

We need to check the predefined variables for auto-fix. Or I think it is good not to auto-fix.

File: src/rules/no-reactive-literals.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
+import type { TSESTree } from "@typescript-eslint/types"
+import { createRule } from "../utils"
+
+const labeledStatementBase = `SvelteReactiveStatement > ExpressionStatement > AssignmentExpression`
+
+export default createRule("no-reactive-literals", {
+  meta: {
+    docs: {
+      description: "Don't assign literal values in reactive statements",
+      category: "Stylistic Issues",
```
Comment (ota-meshi) [inline src/rules/no-reactive-literals.ts:10]: Probably not a stylistic issue. I think this rule should be `"Best Practices"`.

```suggestion
      category: "Best Practices",
```

**Review by ota-meshi — CHANGES_REQUESTED** [review]

File: src/rules/no-reactive-literals.ts (hunk 2 above)
Comment (tivac) [reply src/rules/no-reactive-literals.ts]: Swapped to suggestions.

**Comment by tivac** [comment]
Had to add some infrastructure to make snapshot testing of suggestions possible, will be useful for #201 as well.

File: docs/rules/no-reactive-literals.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
+#  (svelte/no-reactive-literals)
```
Comment (ota-meshi) [inline docs/rules/no-reactive-literals.md:1]: Could you run `yarn update` to regenerate the documentation headers?

**Review by ota-meshi — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**